deepchecks/vision/checks/model_evaluation/weak_segments_performance.py

```diff
@@ -0,0 +1,208 @@
+        for prop_name, prop_value in properties_results.items():
+            self._properties_results[prop_name].extend(prop_value)
+
+        predictions = [tens.detach() for tens in batch.predictions]
+        labels = [tens.detach() for tens in batch.labels]
+        self._sample_scores.extend(self.scorer(predictions, labels))
+
+    def compute(self, context: Context, dataset_kind: DatasetKind) -> CheckResult:
+        """Find the segments with the worst performance."""
+        results_dict = self._properties_results
+        results_dict['score'] = self._sample_scores
+        results_df = pd.DataFrame(results_dict)
+
+        cat_features = [p['name'] for p in self.image_properties if p['output_type'] == 'categorical']
+        num_features = [p['name'] for p in self.image_properties if p['output_type'] == 'numerical']
+        all_features = cat_features + num_features
+
+        dataset = Dataset(results_df, cat_features=cat_features, features=all_features, label='score')
+
+        encoded_dataset = self._target_encode_categorical_features_fill_na(dataset)
+        dummy_model = _DummyModel(test=encoded_dataset, y_pred_test=np.asarray(self._sample_scores),
+                                  y_proba_test=np.asarray(self._sample_scores),
+                                  validate_data_on_predict=False)
+        # the predictions are passed both to pred and proba and each scorer knows which parameter to use.
+        feature_rank = np.asarray(all_features)
+
+        weak_segments = self._weak_segments_search(dummy_model, encoded_dataset, feature_rank,
+                                                   self._sample_scores, self._dummy_scorer)
+        if len(weak_segments) == 0:
+            raise DeepchecksProcessError('WeakSegmentsPerformance was unable to train an error model to find weak '
+                                         'segments. Try increasing n_samples or supply additional properties.')
+
+        avg_score = round(results_df['score'].mean(), 3)
+
+        display = self._create_heatmap_display(dummy_model, encoded_dataset, weak_segments, avg_score,
+                                               self._dummy_scorer) if context.with_display else []
+
+        for idx, segment in weak_segments.copy().iterrows():
+            for feature in ['Feature1', 'Feature2']:
+                if segment[feature] in encoded_dataset.cat_features:
+                    weak_segments[f'{feature} range'][idx] = \
+                        self._format_partition_vec_for_display(segment[f'{feature} range'], segment[feature], None)[0]
+
+        display_msg = 'Showcasing intersections of features with weakest detected segments.<br> The full list of ' \
+                      'weak segments can be observed in the check result value. '
+        return CheckResult({'weak_segments_list': weak_segments, 'avg_score': avg_score, 'scorer_name': 'average_loss'},
```

Comment:
Can do 'scorer_name': f'average {scorer_name}'

Reply:
maybe change the 'avg_score' argument name since it is confusing when the scorer name contains the word average as well. If you decide to change, also change in the tabular check

Reply:
its already a dict.. I suggested changing the keys and values